uffi-compat/uffi-compat.lisp

```diff
@@ -525,24 +525,22 @@ library type if type is not specified."
   #+clisp (sys::getenv (string var))
   #+(or cmu scl) (cdr (assoc (string var) ext:*environment-list* :test #'equalp
                              :key #'string))
-  #+gcl (si:getenv (string var))
+  #+(or ecl gcl) (si:getenv (string var))
   #+lispworks (lw:environment-variable (string var))
   #+lucid (lcl:environment-variable (string var))
   #+(or mcl ccl) (ccl::getenv var)
   #+sbcl (sb-ext:posix-getenv var)
-  #-(or allegro clisp cmu scl gcl lispworks lucid mcl ccl sbcl)
+  #-(or allegro clisp cmu ecl scl gcl lispworks lucid mcl ccl sbcl)
   (error 'not-implemented :proc (list 'getenv var)))
 
 ;; Taken from UFFI's src/os.lisp
 ;; modified from function ASDF -- Copyright Dan Barlow and Contributors
-(defun run-shell-command (control-string  &rest args &key output)
+(defun run-shell-command (control-string &rest args)
   "Interpolate ARGS into CONTROL-STRING as if by FORMAT, and
 synchronously execute the result using a Bourne-compatible shell, with
 output to *trace-output*.  Returns the shell's exit code."
-  (unless output
-    (setq output *trace-output*))
-
-  (let ((command (apply #'format nil control-string args)))
+  (let ((command (apply #'format nil control-string args))
+        (output *trace-output*))
```

Comment:
Hmm. It seems like this never worked properly, right? When :output was passed, it would include that in the command string! That's what you're fixing here, correct?